### ISSUE TITLE
Provide one util  to distinguish bootstrap context and application context when compatible running with Spring Cloud 

### ISSUE THREAD

**Issue by guanchao-yang** [alipay/sofa-boot#232]
### Question
When spring-cloud-context dependency added in classpath, the `org.springframework.context.ApplicationContextInitializer` implementation will be called twice. If we have many ApplicationContextInitializer implementation with Spring Boot ,when spring-cloud-context dependency
added ,our application starter ApplicationContextInitializer run twice.  We should add some identification to distinguish bootstrap context and application context.

> [Reference Issue Here](https://github.com/spring-cloud/spring-cloud-config/issues/1151)

### Scene and advice

Provide one util  to distinguish bootstrap context and application context when compatible running with Spring Cloud. Through this util we can distinguish  bootstrap context loaded by spring cloud and application context loaded by application and expected!

**Comment by QilongZhang** [comment]
Spring  Boot support standard way to distinguish context type. Using `org.springframework.context.ApplicationContext#getId` 
```java
  /**
  * Return the unique id of this application context.
  * @return the unique id of the context, or {@code null} if none
  */
  String getId();
```

And， i support to add a eventListener to remove all SOFABoot initializer， maybe it's a more acceptable way.

**Comment by guanchao-yang** [comment]
@QilongZhang  [The issue](https://github.com/spring-cloud/spring-cloud-config/issues/1151) suggest id but id can be changed by configuration and handled by `ContextIdApplicationContextInitializer ` but if `ApplicationContextInitializer` initialized before `ContextIdApplicationContextInitializer ` ,this way will not tack work. 

>  i support to add an eventListener to remove all SOFABoot initializer， maybe it's a more acceptable way.

All SOFABoot initializer and other initializer only implementation `ApplicationContextInitializer` and we can't distinguish.And Spring Boot initialize the all ApplicationContextInitializers by `org.springframework.boot.SpringApplication#getSpringFactoriesInstances(java.lang.Class<T>, java.lang.Class<?>[], java.lang.Object...)` twice ,there is no event we can listen to change the behavior.

We can discuss other way also.

**Comment by QilongZhang** [comment]
@guanchao-yang  we can listen `ApplicationEnvironmentPreparedEvent` to reset the initializers.  As we only care SOFABoot initializer, we can distinguish them in a special manner, such as package info.

**Comment by guanchao-yang** [comment]
@QilongZhang This way can work and we should provide the specification which restrain the initializers include or exclude , such as start with `com.alipay.sofa.initializers.include` include and start with `com.alipay.sofa.initializers.exclude` exclude or through config file config include and exclude. But all the implementation key point is we should provide one util to distinguish bootstrap context and application context and based on this we can encapsulate.

**Comment by QilongZhang** [comment]
`SOFABootEnvUtils` is used to distinguish spring cloud environment.  It originates from a question that initialization  of `SofaBootRpcInitializer`  in spring cloud triggers initialization of log configuration of `sofa-common-tools`. So we offer `SOFABootEnvUtils` to other starter to check whether execute in spring cloud.  Actually , other than Initializer,  applicationListener or post processor defined in any starter have chance to be invoked spring cloud. We could not remind them of distinguishing spring cloud or spring boot.  Another solution  should be offered.
https://github.com/alipay/sofa-tracer/pull/107

**Comment by QilongZhang** [comment]
fixed in #268
